scripts/OverlaySurvey.py

```diff
@@ -289,28 +318,48 @@ def run_survey(args):
         try:
             SIMULATION = sim.SurveySimulation(args.simGraph, args.simRoot)
         except sim.SimulationError as e:
-            print(f"Error: {e}")
+            logger.critical("%s", e)
             sys.exit(1)
 
     url = args.node
 
     peers = url + "/peers"
-    survey_request = url + "/surveytopology"
+    start_collecting = url + "/startsurveycollecting"
+    stop_collecting = url + "/stopsurveycollecting"
     survey_result = url + "/getsurveyresult"
-    stop_survey = url + "/stopsurvey"
-
-    duration = int(args.duration)
-    params = {'duration': duration}
 
-    # reset survey
-    get_request(url=stop_survey)
+    # start survey recording
+    nonce = random.randint(0, 2**32-1)
+    logger.info("Starting survey with nonce %s", nonce)
+    response = get_request(url=start_collecting, params={'nonce': nonce})
+    if response.text != util.START_SURVEY_COLLECTING_SUCCESS_TEXT:
+        logger.critical("Failed to start survey: %s", response.text)
+        sys.exit(1)
+
+    # Sleep for duration of collecting phase
+    logger.info("Sleeping for collecting phase (%i minutes)", args.duration)
+    time.sleep(args.duration * 60)
```

Comment:
do we need to sleep an extra minute just to allow the start collecting phase message to propagate?

Reply:
[`args.duration` has a minimum value of `1`](https://github.com/bboston7/stellar-core/blob/0d339f979cfbe63e9e23f0350765286734ba5170/scripts/OverlaySurvey.py#L507), so in effect the "start collecting phase" message will have at least 1 minute to propagate.